### DISCUSSION TITLE
Child Node positionAbsolute isn't changing

### DISCUSSION THREAD

**Discussion by AmmarZeer** [xyflow/xyflow#2736] [Q&A]
Nodes can either be on their own or grouped (in a subflow) if it was in a subflow it'll have 2 positioning property:
1) position: which is relative to it's parent (subflow).
2) positionAbsolute: which is relative to the viewport.

When I move the subflow the position of the the subflow (parent) changes correctly but its children positionAbsolute stays the same (even though it should change)

Example: I have these two nodes inside the nodes array, 0 is the parent Node, and 1 is the child Node
Here's the positions when we initialize the flow:
<img width="1361" alt="Screen Shot 2023-01-10 at 1 18 12 PM" src="https://user-images.githubusercontent.com/93482734/211526755-dad4d8b4-bdcc-426f-8bd3-c771a0c5abd0.png">

Now I have moved the parent horizontally thus the change of its absolute position, but the child positionAbsolute is still the same as the old position, shouldn't the nodes state also detect that change?
<img width="1101" alt="Screen Shot 2023-01-10 at 1 18 43 PM" src="https://user-images.githubusercontent.com/93482734/211527062-697bd767-b146-47a2-bd24-26e0352034f3.png">

**Comment by amintabrizi** [comment]
I had the same problem and I solved it in this way
https://codesandbox.io/s/epic-fast-3q9ms9

**Comment by septum** [comment]
Is there a way to get the positions updated in the nodes array?

**Comment by bcakmakoglu** [comment]
Where are you accessing the nodes from?

I don't see any issue with the position update in [this sandbox](https://codesandbox.io/s/zen-knuth-sj2j3j?file=/App.js).
When I move the parent, the childs position absolute is updated and changed.

**Reply by septum** [reply]
I was accessing them directly as `nodes` from an [external store](https://reactflow.dev/docs/guides/state-management/), but thanks to this [response](https://github.com/wbkd/react-flow/issues/2603#issuecomment-1571745327), I am now accessing them from `nodeInternals` which makes me think is the same as `instance.getNodes()`.

In sort of a tangent, which would be the more performant way to access them?
